File: leetcode/easy/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # one pass O(N) time O(N) space
        d = {}
        for i in range(len(nums)):
            n = nums[i]
            if n in d: return [d[n], i]
            d[target - n] = i

File: leetcode/easy/twoSum.cpp
#include <unordered_map>
vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> m;
    for (int z = 0; z < nums.size(); z++) {
        int n = nums[z];
        if (m.find(target - n) != m.end())
            return {z, m[target-n]};
        m[n] = z;
    }
} 
